client/flutter/lib/pages/main_pages/app_tab_router.dart

```diff
@@ -40,6 +41,52 @@ class AppTabRouter extends StatelessWidget {
 
   AppTabRouter(this.tabs, this.navItems);
 
+  @override
+  _AppTabRouterState createState() => _AppTabRouterState(tabs, navItems);
+}
+
+class _AppTabRouterState extends State<AppTabRouter> {
+  CupertinoTabController _controller;
+  FirebaseAnalytics _analytics;
+  int lastTabIndex;
+  @override
+  void initState() {
+    _analytics = FirebaseAnalytics();
+    _controller = CupertinoTabController();
+    lastTabIndex = _controller.index;
+
+    _controller.addListener(
+      () {
+        int currentIndex = _controller.index;
+
+        String lastTab = tabTitle(lastTabIndex);
+        String currentTab = tabTitle(currentIndex);
+
+        lastTabIndex = currentIndex;
+
+        _analytics.logEvent(
+          name: 'changed_tabs',
+          parameters: {
+            'current_tab': currentTab,
+            'last_tab': lastTab,
+          },
+        );
+      },
+    );
+    super.initState();
```

Comment:
It's generally good practice to put your superclass calls to the method you're overriding at the beginning of your method, just in case the superclass changes things you're setting up here or initializes things required for your code. Please move the `super.initState();` to the first line in this method

Reply:
done